src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -115,13 +124,16 @@ public bool ExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionConte
         }
 
         private void CodeCleanupOrFormat(
-            FormatDocumentCommandArgs args, Document document,
+            FormatDocumentCommandArgs args, Document document, ICodeCleanupService codeCleanupService,
             IProgressTracker progressTracker, CancellationToken cancellationToken)
         {
-            var codeCleanupService = document.GetLanguageService<ICodeCleanupService>();
+            var workspace = document.Project.Solution.Workspace;
+            ABTest(document, workspace);
 
-            var docOptions = document.GetOptionsAsync(cancellationToken).WaitAndGetResult(cancellationToken);
-            if (docOptions.GetOption(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting))
+            // Show different gold bar text depends on PerformAdditionalCodeCleanupDuringFormatting value
+            ShowGoldBarForCodeCleanupConfigurationIfNeeded(document, workspace.Options.GetOption(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting, document.Project.Language));
```

Comment:
This function skips showing the gold bar if "NeverShowCodeCleanupInfoBarAgain" is set. So if somebody previously saw the gold bar, clicked "never show again", and we move them to the test group, we'll change their value but not show the bar at all? That seems bad.